schema/bom-1.7.proto

```diff
@@ -106,8 +106,17 @@ message Component {
   optional string group = 7;
   // The name of the component. This will often be a shortened, single name of the component. Examples: commons-lang3 and jquery
   string name = 8;
-  // The component version. The version should ideally comply with semantic versioning but is not enforced. Version was made optional in v1.4 of the spec. For backward compatibility, it is recommended to use an empty string to represent components without version information.
-  string version = 9;
+  oneof version_choice {
+    // The component version. The version should ideally comply with semantic versioning but is not enforced.
+    // Version was made optional in v1.4 of the spec.
+    // For backward compatibility, it is recommended to use an empty string to represent components without version information.
+    string version = 9;
```

Comment:
> error: Field "9" with name "version" on message "Component" moved from outside to inside a oneof.

is this a breaking change? need to investigate further.

from what I've read so far, this "oneOf" makes "version" optional, now.
and this might be a breaking change on the wire, as the value was required before.

Reply:
@nscuro @andreas-hilti, may I ask for your opinion on this change?

Reply:
Was version really required? The [Protobuf documentation for version 1.6](https://cyclonedx.org/docs/1.6/proto/#cyclonedx.v1_6.Component) marks it as optional and adds "Version was made optional in v1.4 of the spec. For backward compatibility, it is recommended to use an empty string to represent components without version information."

Reply:
> Was version really required?
> The [Protobuf documentation for version 1.6](https://cyclonedx.org/docs/1.6/proto/#cyclonedx.v1_6.Component) marks it as optional [...]

marks it optional per comment/spec. it is still required by protobuf


yes. version was required before CDX1.4.
the protobuf implementation was introduced in 1.3.

for backwards compatibility of protobuf (over the wire) the “version” field was kept required up until today.

Reply:
reverted the `choice`, and went back to simple fileds.